src/state/sagas/datasets/DeleteDataset/DeleteDataset.js

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import { takeEvery, call, put } from 'redux-saga/effects';
+import { t } from 'i18next';
+import { Api } from '../../../../services/config/Api';
+import { fetchAllDatasetsData } from '../FindAllDatasets/FindAllDatasets';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+import { STATUSES } from '../../../commons/Constants';
+
+export function* deleteDataset(action) {
+  try {
+    const organizationId = action.organizationId;
+    const datasetId = action.datasetId;
+
+    yield call(Api.Datasets.deleteDataset, organizationId, datasetId);
+    yield call(fetchAllDatasetsData, organizationId);
```

Comment:
I'm not a huge fan of fetching all datasets again after deleting one of them, it will have a noticeable impact on perfs :sweat_smile: 
If we change the `DELETE_DATASET` action to remove items from the datasets list, we no longer need to fetch the possible thousands of datasets after each dataset removal.

Note: I think we're doing the same when deleting scenarios, but we probably don't have to ("c'est historique" :grin: )